### PR TITLE
Remove double to string convertion in FormData

### PR DESCRIPTION
**Pull request by doochik**
These convertions introduced by 2 different PRs
https://github.com/facebook/react-native/commit/5061fde3173e6228d544b5df1ce8ce4e2a7dd2a4
https://github.com/facebook/react-native/commit/de392b5351494675220e0ef5176d8a8f3d473477

### CODE REVIEW

**Comment by olegbl** [comment]
Make sure it works for booleans, etc... Not every non-string is an object. (And { ...value } crashes when value is a boolean.)

**Comment by facebook-github-bot-6** [comment]
@olegbl would you mind taking a look at this pull request? It's been a while since the last commit was reviewed.

File: Libraries/Network/FormData.js [hunk 1]
```diff
@@ -85,7 +81,7 @@ class FormData {
         }
         return {...value, headers, fieldName: name};
       }
-      // Cast to string all other values
+      // Convert non-object values to strings as per FormData.append() spec
       return {string: String(value), headers, fieldName: name};
```
Comment (satya164) [inline Libraries/Network/FormData.js]: We should probably remove this extra conversion instead of removing the one at the top

Comment (doochik) [inline Libraries/Network/FormData.js]: why? there are identically

Comment (satya164) [inline Libraries/Network/FormData.js]: Check @olegbl 's comment

**Comment by doochik** [comment]
@olegbl @satya164 
Do you mean cases like `formData.append('foo', new Boolean(true))`?
No, there is no crashes. Just value is empty in `formData.getParts()` cause no iteratable props in `new Boolean()`.

I can rewrite this https://github.com/facebook/react-native/blob/bef175af61d387084fd9161e495cac3e1b414556/Libraries/Network/FormData.js#L79 to something like `if (isPlainObject(value)) { ... }`. But I think it's overhead.

In this PR i've just removed some duplicate code
* value casting at line 69: `value = '' + value` and line 89: `value: String(value)`
* checking value type at line 68: `if (typeof value !== 'object')` and line 79: `if (typeof value === 'object')`).

I've removed first conversion (`value = '' + value`) cause `String(value)` is more obviously in my opinion.

**Comment by olegbl** [comment]
I meant formData.append('foo', true). Looks like it should be fine since "if (typeof value === 'object') {" got added some time back.

**Comment by doochik** [comment]
Yep, it should be fine and will be casted to `foo=true` in HTTP request

**Comment by satya164** [comment]
Can you rebase against master so that the tests pass?

**Comment by doochik** [comment]
i've rebased my branch

**Comment by facebook-github-bot-3** [comment]
@doochik do you have any updates for this pull request? It's been a while so we wanted to check in and see if you've looked at the requested changes.

**Comment by satya164** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**